File: official/docs/ruby/v5/carbon-offset/one-call-buy.rb
require 'easypost'

client = EasyPost::Client.new(api_key: ENV['EASYPOST_API_KEY'])

shipment = client.shipment.create(
  {
    carrier_accounts: ['ca_...'],
    to_address: {
      name: 'Dr. Steve Brule',
      street1: '179 N Harbor Dr',
      city: 'Redondo Beach',
      state: 'CA',
      zip: '90277',
      country: 'US',
      phone: '4155559999',
      email: 'dr_steve_brule@gmail.com',
    },
    from_address: {
      name: 'EasyPost',
      street1: '417 Montgomery Street',
      street2: '5th Floor',
      city: 'San Francisco',
      state: 'CA',
      zip: '94104',
      country: 'US',
      phone: '4153334445',
      email: 'support@easypost.com',
    },
    parcel: {
      length: 20.2,
      width: 10.9,
      height: 5,
      weight: 65.9,
    },
  },
  true,
)

puts shipment

File: official/docs/golang/current/luma/one-call-buy.go
package example

import (
  "fmt"

  "github.com/EasyPost/easypost-go/v5"
)

func oneCallBuyLuma() {
  client := easypost.New("EASYPOST_API_KEY")

  params := map[string]any{
    "to_address": map[string]any{
      "name":    "Dr. Steve Brule",
      "street1": "5744 Silverton Ave",
      "city":    "McKinney",
      "state":   "TX",
      "zip":     "75070",
      "country": "US",
      "phone":   "8573875756",
      "email":   "dr_steve_brule@gmail.com",
    },
    "from_address": map[string]any{
      "name":    "EasyPost",
      "street1": "417 Montgomery Street",
      "street2": "5th Floor",
      "city":    "San Francisco",
      "state":   "CA",
      "zip":     "94104",
      "country": "US",
      "phone":   "4153334445",
      "email":   "support@easypost.com",
    },
    "parcel": map[string]any{
      "length": 20.2,
      "width":  10.9,
      "height": 5.0,
      "weight": 65.9,
    },
    "carrier_accounts":  []string{"ca_..."},
    "persist_label":      true,
    "ruleset_name":       "required_deliver_by_date",
    "planned_ship_date":  "2025-07-21",
    "deliver_by_date":    "2025-07-23",
    "insurance":          "50.00",
  }

  shipment, _ := client.Shipment.CreateAndBuyLuma(params)

  fmt.Println(shipment)
}
